### PR TITLE
add options for wart remover

### PR DESCRIPTION
**Pull request by winitzki**

### CODE REVIEW

File: lib/src/main/scala/code/winitzki/jc/StaticAnalysis.scala [hunk 1]
```diff
@@ -165,10 +166,10 @@ private object StaticAnalysis {
     val possibleDeadlocks: Seq[(OutputMoleculeInfo, List[OutputMoleculeInfo])] =
       reactions.flatMap(_.info.outputs)
         .flatMap {
-          _.tails.filter {
-            case t :: ts => t.molecule.isBlocking
-            case Nil => false
-          }.map { l => (l.head, l.tail) } // The filter above guarantees that `l` is non-empty now.
+          _.tails.map {
```
Comment (phderome) [inline lib/src/main/scala/code/winitzki/jc/StaticAnalysis.scala]: map.flatten normally becomes flatMap. Maybe flatMap cannot do pattern matching.

**Review by phderome — COMMENTED** [review]
lgtm

**Comment by winitzki** [comment]
There are still some problems when I changed `Molecule#toString`. Need to revert that.

File: lib/src/main/scala/code/winitzki/jc/StaticAnalysis.scala (hunk 1 above)
Comment (winitzki) [reply lib/src/main/scala/code/winitzki/jc/StaticAnalysis.scala]: Yes, it can! I will replace this by a `flatMap`.

**Comment by winitzki** [comment]
The wart remover was useful: it uncovered a few potential problems with the code. But most of its warnings are too stringent for this kind of code. I can't stop using OptionToIterable or Any.

**Pull request merged**